src/sds.c

```diff
@@ -281,6 +285,17 @@ sds sdsMakeRoomFor(sds s, size_t addlen) {
     return s;
 }
 
+/* Enlarge the free space at the end of the sds string more than needed,
+ * pre-allocate at least `sdslen(s) + addlen * 2` bytes of memory. */
```

Comment:
is that technically right?
i don't think we'll always add `addlen * 2`. addlen can be 10mb, and we'll only add 1mb.
i suppose the comment can be vague, just saying we preallocate for future growth and avoid repeated re-allocations. 

```suggestion
 * This is useful to avoid repeated re-allocations when repeatedly appending to the sds. */
```